go/vt/vttablet/tabletserver/schema/historian.go

```diff
@@ -20,49 +20,55 @@ import (
 	"context"
 	"sort"
 	"sync"
+	"time"
+
+	"vitess.io/vitess/go/vt/sidecardb"
 
 	"vitess.io/vitess/go/mysql"
 	"vitess.io/vitess/go/sqltypes"
 	"vitess.io/vitess/go/vt/log"
 	binlogdatapb "vitess.io/vitess/go/vt/proto/binlogdata"
-	"vitess.io/vitess/go/vt/sidecardb"
 	"vitess.io/vitess/go/vt/vtgate/evalengine"
 	"vitess.io/vitess/go/vt/vttablet/tabletserver/connpool"
 	"vitess.io/vitess/go/vt/vttablet/tabletserver/tabletenv"
 
 	"vitess.io/vitess/go/vt/sqlparser"
 )
 
-const getSchemaVersions = "select id, pos, ddl, time_updated, schemax from %s.schema_version where id > %d order by id asc"
+const getInitialSchemaVersions = "select id, pos, ddl, time_updated, schemax from %s.schema_version where from_unixtime(time_updated) > %d order by id asc"
```

Comment:
We still don't have any tests -- unit or endtoend -- for this part, do we? Something that would have caught the previous issue and will confirm that this works as expected with various inputs. I think we should figure out a way to add a test for it -- with a preference for unit test. I can help with this if you need to bounce some ideas around.

Reply:
Sounds good. I'll see what I can do and will reach out if I run into any issues.